### PR TITLE
CYB-409 - Engagement index: only 10 of 47 users appear on index initially

### PR DESCRIPTION
**Pull request by nicolaasmatthijs**
Page the query to ensure that all users are pre-loaded

https://jira.ets.berkeley.edu/jira/browse/CYB-409

### CODE REVIEW

**Comment by nicolaasmatthijs** [comment]
The build succeeded. Ready for review again.

File: app/services/canvas/students_processor.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  def call(students)
+    if students.respond_to?(:each)
+      students.each do |student|
+        student_attributes = {}
+        student_attributes[:name] = student["name"]
+        student_attributes[:sortable_name] = student["sortable_name"]
+        student_attributes[:canvas_user_id] = student["id"].to_i
+        student_attributes[:sis_user_id] = student["sis_user_id"] || -1
+        student_attributes[:section] = "Unknown"
+
```
Comment (johncrossman) [inline app/services/canvas/students_processor.rb]: Only put phrases in double quotes when you want string interpolation.  The snippet *student["sortable_name"]* should be *student['sortable_name']*

File: app/services/canvas/students_processor.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        # we can't use #find_or_create_by as if any other attributes differ but canvas_user_id value is already
+        # present, ActiveRecord will not match, and will try to create a new record, violating uniqueness on that key.
+        student_record = Student.where({canvas_user_id: student_attributes[:canvas_user_id]}).first
+        if student_record.nil?
+          Student.create(student_attributes)
+        else
+          student_record.update_attributes(student_attributes)
```
Comment (johncrossman) [inline app/services/canvas/students_processor.rb]: our coding style is to exclude parentheses when it's a single arg to the method. So *student_record.update_attributes(student_attributes)*   becomes   *student_record.update_attributes student_attributes*

**Comment by johncrossman** [comment]
@nicolaasmatthijs   I dropped in a couple style issues. Once those are fixed I'll merge. Thanks!

File: app/services/canvas/students_processor.rb (hunk 2 above)
Comment (nicolaasmatthijs) [inline app/services/canvas/students_processor.rb]: This is existing code that was moved from a different file. The Data Cultures code uses the parentheses approach consistently across its entire codebase, so I'd prefer to keep it as is to not generate inconsistencies

**Comment by nicolaasmatthijs** [comment]
Followed up on the first comment and responded to the second one.

File: app/services/canvas/students_processor.rb (hunk 2 above)
Comment (johncrossman) [inline app/services/canvas/students_processor.rb]: Makes sense

**Comment by nicolaasmatthijs** [comment]
The build passed. Ready for more review and merging.

**Comment by johncrossman** [comment]
Ready for merge.

**Pull request merged**